src/servers/trigger.ts

```diff
@@ -107,13 +107,18 @@ export async function onchainTransactionEventHandler(tx) {
   } else {
     // incoming transaction
 
-    let user
+    let wallet: Wallet
     try {
-      user = await User.findOne(
-        { "onchain.address": { $in: tx.output_addresses } },
-        { lastIPs: 0, lastConnection: 0 },
-      )
-      if (!user) {
+      const walletsRepo = WalletsRepository()
+
+      // TODO: tx.output_addresses pass an array of address
+      // in the unlikely event multiple destination addresses are belong to the Galoy wallet
+      // only the first one will be notified
+      const wallet_ = await walletsRepo.findByAddress(tx.output_addresses)
+      if (wallet_ instanceof Error) return
+      wallet = wallet_
+
+      if (!wallet) {
```

Comment:
this validation is not necessary, if address is not assigned to an user wallet will be an instance of Error